packages/ui/src/components/Stepper/Step.tsx

```diff
@@ -0,0 +1,247 @@
+import { css, keyframes } from '@emotion/react'
+import styled from '@emotion/styled'
+import type { ReactNode } from 'react'
+import { useMemo } from 'react'
+import { Bullet } from '../Bullet'
+import { Stack } from '../Stack'
+import { Text } from '../Text'
+import { useStepper } from './StepperProvider'
+
+const LINE_HEIGHT_SIZES = {
+  small: 2,
+  medium: 4,
+} as const
+
+type StepProps = {
+  onClick?: (index: number) => void
+  /**
+   * Automatically attribued by the parent Stepper
+   */
+  index?: number
+  /**
+   * Whether the step is disabled
+   */
+  disabled?: boolean
+  /**
+   * Title of the step
+   */
+  title?: ReactNode
+  /**
+   * For additional information.
+   * Use prop `title` to properly name the step
+   */
+  children?: ReactNode
+  className?: string
+  'data-testid'?: string
+}
+const loadingAnimation = (size: 'small' | 'medium') => keyframes`
+  from {
+    width: 0;
+  }
+  to {
+    width: calc(
+      100% - ${size === 'small' ? '24px' : '32px'} -
+       8px}
+    );
+  }
+`
+
+const loadingStyle = (size: 'small' | 'medium') => css`
+  animation: ${loadingAnimation(size)} 1s linear infinite;
+`
+
+const StyledBullet = styled(Bullet)<{
+  size: 'small' | 'medium'
+  isActive: boolean
+}>`
+  margin-top: ${({ theme, size }) =>
+    size === 'small' ? theme.space['0.5'] : 0};
+  transition: box-shadow 300ms;
+  min-width: ${({ theme, size }) =>
+    size === 'small' ? theme.space[3] : theme.space[4]};
+  ${({ theme, isActive }) =>
+    isActive
+      ? `background-color: ${theme.colors.primary.backgroundStrongHover}; 
+          box-shadow: ${theme.shadows.focusPrimary};`
+      : null};
+`
+
+const StyledText = styled(Text)<{ labelPosition: 'right' | 'bottom' }>`
+  margin-top: ${({ theme }) => theme.space['1']};
+  transition: text-decoration-color 250ms ease-out;
+  text-decoration-thickness: 1px;
+  text-underline-offset: 2px;
+  text-decoration-color: transparent;
+  white-space: normal;
+`
+
+const StyledStepContainer = styled(Stack)<{
+  isActive: boolean
+  isDone: boolean
```

Comment:
```suggestion
  'data-is-active': boolean
  'data-is-done': boolean
```
data-attributes won't re-compute the style and allow us to get static file